Adding a code sample of how the modal widget can be initialized

eduard13 opened the pull request:
## Purpose of this PR

This PR is an improvement of Modal widget, which provides an example of how the modal can be used and how the widget's options can be passed during initialization. Also the example's result preview is provided.
Moreover, an additional (important) widget's option was added to the documentation.

## Affected URLs

<!-- REQUIRED List the URLs you are changing. Not needed for large numbers of files. -->

- https://devdocs.magento.com/guides/v2.1/javascript-dev-guide/widgets/widget_modal.html
- https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/widgets/widget_modal.html
- https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html

whatsnew
Added a [new code sample](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html#code-sample) showing how to initialize a modal widget and pass options to it.

devops-devdocs commented:
An admin must run tests on this PR before it can be merged.

rogyar commented:
CodeBase: https://github.com/magento/magento2/blob/2.1/app/code/Magento/Ui/view/base/web/js/modal/modal.js#L73

rogyar commented on guides/v2.1/javascript-dev-guide/widgets/widget_modal.md:
```diff
@@ -202,6 +209,36 @@ Called when the modal window is opened.
+
+The result is a modal and a button (_Click Here_) which opens the modal.
+
+![Tabs Widget]({{ page.baseurl }}/javascript-dev-guide/widgets/images/modal-widget-result.png)
```
Please, double check the `[Tabs Widget]` title here. The picture is about a modal widget example.

jeff-matthews commented on guides/v2.1/javascript-dev-guide/widgets/widget_modal.md:
```diff
@@ -138,6 +139,12 @@ Translated title for popup window.
 **Default value**: empty
 
+### `trigger` {#modal_trigger}
+The element which triggers the modal.
```
```suggestion
The element that triggers the modal.
```

jeff-matthews commented on guides/v2.1/javascript-dev-guide/widgets/widget_modal.md:
```diff
@@ -202,6 +209,36 @@ Called when the modal window is opened.
+
+### Result
+
+The result is a modal and a button (_Click Here_) which opens the modal.
```
```suggestion
The result is a modal and a button (_Click Here_) that opens the modal.
```

eduard13 commented:
Hi @jeff-matthews, the suggested changes were applied.
Thank you for the review.

eduard13 commented:
@jeff-matthews, yes they are. Interesting, I didn't know about that, thank you for noticing.
Please note that I moved it into the mentioned folder, optimized it and removed the other ones.
Thank you.

jeff-matthews commented:
Perfect @eduard13! Thank you for taking the extra step to optimize the image as well. We haven't had good contributor guidelines around doing that in the past, but we'll create some soon. Thank you for being proactive.

jeff-matthews commented:
running tests

jeff-matthews commented:
Tests failed. Here's the log:

```
+ rake test:cicd
[35mChecking links with htmlproofer...[0m
Running ["LinkChecker::DoubleSlashCheck", "ImageCheck", "LinkCheck", "ScriptCheck"] on ["_site"] on *.html... 

Ran on 3824 files!

- internal image /common/images/js/widgets/modal-widget-result.png does not exist
  *  _site/guides/v2.1/javascript-dev-guide/widgets/widget_modal.html (line 2368)
  *  _site/guides/v2.2/javascript-dev-guide/widgets/widget_modal.html (line 2430)
  *  _site/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html (line 2486)
rake aborted!
HTML-Proofer found 3 failures!```

jeff-matthews commented on guides/v2.1/javascript-dev-guide/widgets/widget_modal.md:
```diff
@@ -202,6 +209,36 @@ Called when the modal window is opened.
+
+The result is a modal and a button (_Click Here_) that opens the modal.
+
+![Modal Widget]({{ site.baseurl }}/common/images/js/widgets/modal-widget-result.png)
```
```suggestion
![Modal Widget]({{ site.baseurl }}/common/images/js/widget/modal-widget-result.png)
```

jeff-matthews replied:
```suggestion
![Modal Widget]({{ site.baseurl }}/common/images/js/widget/modal-widget-result.png)
```

jeff-matthews commented:
running tests

eduard13 commented:
@jeff-matthews now it should be fine I guess.

jeff-matthews commented:
Yes, I think so too. Almost finished running tests.

jeff-matthews commented:
running tests
